crates/aptos-logger/src/metadata.rs

```diff
@@ -60,68 +60,78 @@ impl Metadata {
+#[derive(
+    Copy,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Debug,
+    Hash,
+    Serialize,
+    Deserialize,
+    FromRepr,
+    EnumString,
+    Display,
+)]
 #[serde(rename_all = "UPPERCASE")]
 pub enum Level {
     /// The "error" level.
     ///
     /// Designates very serious errors.
+    #[strum(ascii_case_insensitive)]
+    #[strum(to_string = "ERROR")]
     Error = 0,
     /// The "warn" level.
     ///
     /// Designates hazardous situations.
-    Warn,
+    #[strum(ascii_case_insensitive)]
+    #[strum(to_string = "WARN")]
+    Warn = 1,
     /// The "info" level.
     ///
     /// Designates useful information.
-    Info,
+    #[strum(ascii_case_insensitive)]
+    #[strum(to_string = "INFO")]
+    Info = 2,
     /// The "debug" level.
     ///
     /// Designates lower priority information.
-    Debug,
+    #[strum(ascii_case_insensitive)]
+    #[strum(to_string = "DEBUG")]
+    Debug = 3,
     /// The "trace" level.
     ///
     /// Designates very low priority, often extremely verbose, information.
-    Trace,
+    #[strum(ascii_case_insensitive)]
+    #[strum(to_string = "TRACE")]
+    Trace = 4,
 }
 
-impl Level {
-    fn from_usize(idx: usize) -> Option<Self> {
-        let lvl = match idx {
-            0 => Level::Error,
-            1 => Level::Warn,
-            2 => Level::Info,
-            3 => Level::Debug,
-            4 => Level::Trace,
-            _ => return None,
-        };
-
-        Some(lvl)
-    }
-}
-
-/// An error given when no `Level` matches the inputted string
-#[derive(Debug)]
-pub struct LevelParseError;
+#[cfg(test)]
+mod test {
+    use super::*;
+    use std::str::FromStr;
 
-impl FromStr for Level {
-    type Err = LevelParseError;
+    #[test]
+    fn test_log_level_from_string() {
+        assert_eq!(Level::Error, Level::from_str("ERROR").unwrap());
+        assert_eq!(Level::Error, Level::from_str("Error").unwrap());
+        assert_eq!(Level::Error, Level::from_str("error").unwrap());
 
-    fn from_str(level: &str) -> Result<Level, Self::Err> {
-        LOG_LEVEL_NAMES
-            .iter()
-            .position(|name| name.eq_ignore_ascii_case(level))
-            .map(|idx| Level::from_usize(idx).unwrap())
-            .ok_or(LevelParseError)
+        assert!(Level::from_str("ERR").is_err());
+        assert!(Level::from_str("err_or").is_err());
     }
-}
 
-impl fmt::Display for Level {
-    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
-        fmt.pad(LOG_LEVEL_NAMES[*self as usize])
+    #[test]
+    fn test_log_level_to_string() {
+        assert_eq!(String::from("ERROR"), Level::Error.to_string());
+        assert_eq!(String::from("WARN"), Level::Warn.to_string());
+        assert_eq!(String::from("INFO"), Level::Info.to_string());
+        assert_eq!(String::from("DEBUG"), Level::Debug.to_string());
+        assert_eq!(String::from("TRACE"), Level::Trace.to_string());
```

Comment:
can you add one more even with a single lower-case for other (non-error) ones too?